MinGW build fails using asm()

As of building with Perl 5.40 using MinGW we are now seeing this issue, it does not occur using 5.38:


```
In file included from factor.c:15:
mulmod.h: In function '_mulmod':
mulmod.h:26:5: warning: implicit declaration of function 'asm' [-Wimplicit-function-declaration]
26 | asm ("mulq %3\n\t" /* mul a*b -> rdx:rax */
| ^~~
mulmod.h:27:23: error: expected ')' before ':' token
27 | "divq %4\n\t" /* (a*b)/c -> quot in rax remainder in rdx */
| ^
| )
28 | :"=a"(dummy), "=&d"(d) /* output */
| ~
mulmod.h:26:9: note: to match this '('
26 | asm ("mulq %3\n\t" /* mul a*b -> rdx:rax */
| ^
mulmod.h: In function '_addmod':
mulmod.h:47:28: error: expected ')' before ':' token
47 | "cmovc %1, %0\n\t" /* if (carry) a := t */
| ^
| )
48 | :"+r" (a), "+&r" (t)
| ~
mulmod.h:46:9: note: to match this '('
46 | asm ("add %2, %1\n\t" /* t := t + b */
| ^
In file included from factor.c:18:
montmath.h: In function '_mulredc63':
montmath.h:51:33: error: expected ')' before ':' token
51 | "addq %%rdx, %%rax \n\t"
| ^
| )
52 | : "=a"(a)
| ~
montmath.h:41:8: note: to match this '('
41 | asm("mulq %2 \n\t"
| ^
montmath.h: In function '_mulredc64':
montmath.h:70:33: error: expected ')' before ':' token
70 | "addq %%rdx, %%rax \n\t"
| ^
| )
71 | : "+&a"(a)
| ~
montmath.h:58:8: note: to match this '('
58 | asm("mulq %1 \n\t"
| ^
montmath.h: In function '_u64div':
montmath.h:89:16: error: expected ')' before ':' token
89 | asm("divq %4"
| ~ ^
| )
90 | : "=a"(c), "=d"(n)
| ~
gmake: *** [Makefile:422: factor.o] Error 1

```

The fix that is working for me is to replace `asm ()` with `__asm__ ()`  -- this is working for gcc on linux, and mingw, but I don't think will work for MSVC.

--- a/montmath.h
+++ b/montmath.h
@@ -38,7 +38,7 @@ static INLINE uint64_t mont_inverse(const uint64_t n) {
 
 /* MULREDC asm from Ben Buhrow */
 static INLINE uint64_t _mulredc63(uint64_t a, uint64_t b, uint64_t n, uint64_t npi) {
-    asm("mulq %2 \n\t"
+    __asm__ ("mulq %2 \n\t"
         "movq %%rax, %%r10 \n\t"
         "movq %%rdx, %%r11 \n\t"
         "mulq %3 \n\t"
@@ -55,7 +55,7 @@ static INLINE uint64_t _mulredc63(uint64_t a, uint64_t b, uint64_t n, uint64_t n
   return a;
 }
 static INLINE uint64_t _mulredc64(uint64_t a, uint64_t b, uint64_t n, uint64_t npi) {
-    asm("mulq %1 \n\t"
+    __asm__ ("mulq %1 \n\t"
         "movq %%rax, %%r10 \n\t"
         "movq %%rdx, %%r11 \n\t"
         "movq $0, %%r12 \n\t"

--- a/mulmod.h
+++ b/mulmod.h
@@ -23,7 +23,7 @@
   /* Beware: if (a*b)/c > 2^64, there will be an FP exception */
   static INLINE UV _mulmod(UV a, UV b, UV n) {
     UV d, dummy;                    /* d will get a*b mod c */
-    asm ("mulq %3\n\t"              /* mul a*b -> rdx:rax */
+    __asm__ ("mulq %3\n\t"              /* mul a*b -> rdx:rax */
          "divq %4\n\t"              /* (a*b)/c -> quot in rax remainder in rdx */
          :"=a"(dummy), "=&d"(d)     /* output */
          :"a"(a), "r"(b), "r"(n)    /* input */
@@ -43,7 +43,7 @@
   static INLINE UV _addmod(UV a, UV b, UV n) {
     UV t = a-n;
     a += b;
-    asm ("add %2, %1\n\t"    /* t := t + b */
+    __asm__ ("add %2, %1\n\t"    /* t := t + b */
          "cmovc %1, %0\n\t"  /* if (carry) a := t */
          :"+r" (a), "+&r" (t)
          :"r" (b)

